### PR TITLE
feat: reorganize set-env-vars step to use python script

### PR DESCRIPTION
**Pull request by kseniyashaydurova**
This PR contains reorganized `set-env-vars` step for buildpack, buildpack-dotnet and buildpack-frontend tasks. 

The difference is in usage python `toml` packages to parse `project.toml` file inside project dir to avoid setting such env vars from terraform code (more info https://github.com/saritasa-nest/saritasa-devops-tekton/issues/8)

### CODE REVIEW

**Comment by kseniyashaydurova** [comment]
@dmitry-mightydevops Current solution works, but I suggest to make a python script, which will take `source_path` and `env_dir` arguments, so we would call it from linux terminal. So we will have no code repeat and we would be able to use our common approach for `slack error notification` (but to make it work we will need to use centos image instead of alpine)

**Comment by dmitry-mightydevops** [comment]
> @dmitry-mightydevops Current solution works, but I suggest to make a python script, which will take `source_path` and `env_dir` arguments, so we would call it from linux terminal. So we will have no code repeat and we would be able to use our common approach for `slack error notification` (but to make it work we will need to use centos image instead of alpine)

Not sure I understand - can you elaborate on it more?

File: tekton-pipelines/base/tasks/buildpack-task.yaml [hunk 1]
```diff
@@ -91,79 +91,57 @@ spec:
...
+          "DOTNET", "RUNTIME_VERSION", "SDK_LOCATION",
+          # paketo go buildpacks
+          "DEPCACHEDIR",
+          # paketo nodejs buildpacks
+          "NODE_", "MEMORY_",
+          # paketo php buildpacks
+          "APP_", "SERVER_", "MIBDIRS", "PHP_",
+          # paketo ruby buildpacks
+          "GEM_", "BUNDLE_"
+        ]
```
Comment (dmitry-mightydevops) [inline tekton-pipelines/base/tasks/buildpack-task.yaml:117]: We don't need it - as new env vars can appear in all the buildpacks. I did it this way to separate from in-POD env vars. You don't need it as you need to take project.toml as the only source of truth. So whatever the env vars are defined in **project.toml** - you persist them on disk.

File: tekton-pipelines/base/tasks/buildpack-task.yaml [hunk 2]
```diff
@@ -91,79 +91,57 @@ spec:
...
+
+        try:
+          with open(source_path, "r") as f:
+            parsed = toml.load(f)
```
Comment (dmitry-mightydevops) [inline tekton-pipelines/base/tasks/buildpack-task.yaml:121]: let's replace var `parsed` with `settings` as it's more semantic.

File: tekton-pipelines/base/tasks/buildpack-task.yaml [hunk 3]
```diff
@@ -91,79 +91,57 @@ spec:
...
+        except FileNotFoundError:
+          quit()
+        except Exception as e:
+          raise Exception("Toml file couldn't be parsed\nTraceback: {e}")
```
Comment (dmitry-mightydevops) [inline tekton-pipelines/base/tasks/buildpack-task.yaml:125]: Unable to parse project.toml file.

File: tekton-pipelines/base/tasks/buildpack-task.yaml [hunk 4]
```diff
@@ -91,79 +91,57 @@ spec:
...
+        env_vars = {item["name"]: item["value"] for item in env_vars}
+
+        # create directory if not exists
+        if env_vars and not os.path.exists(env_dir):
+            os.makedirs(env_dir)
+
+        # create env variables files
+        regex = f"^({'|'.join(vars_prefixes)})"
+        for name, value in env_vars.items():
+          if re.match(regex, name):
```
Comment (dmitry-mightydevops) [inline tekton-pipelines/base/tasks/buildpack-task.yaml:137]: no need to match by regexp. See comments above.

Comment (dmitry-mightydevops) [reply tekton-pipelines/base/tasks/buildpack-task.yaml:137]: let's apply the same changes to others below.

**Review by dmitry-mightydevops — CHANGES_REQUESTED** [review]

File: tekton-pipelines/base/kustomization.yaml [hunk 5]
```diff
@@ -11,6 +11,7 @@ commonLabels:
   deploy: argocd
 
 resources:
+  - tasks/set-env-vars-task.yaml
```
Comment (dmitry-mightydevops) [inline tekton-pipelines/base/kustomization.yaml:14]: buildpack-env-vars-task.yaml

**Review by dmitry-mightydevops — APPROVED** [review]
# Love it!

**Review by roman-mightydevops — APPROVED** [review]

**Review by populov — APPROVED** [review]

File: tekton-pipelines/buildpacks-dotnet/tasks/build.yaml [hunk 6]
```diff
@@ -123,6 +123,71 @@ spec:
...
+        except Exception:
+          raise Exception("Toml file couldn't be parsed")
+
+      volumeMounts:
+        - name: layers-dir
+          mountPath: /layers
+        - name: $(params.platform_dir)
+          mountPath: /platform
+      securityContext:
+        privileged: true
```
[the remark this answers is not in the document]
Comment (kseniyashaydurova) [reply tekton-pipelines/buildpacks-dotnet/tasks/build.yaml]: @populov I haven't created `dotnet` build, so I don't know why it was added. Please discuss it with @dmitry-mightydevops

**Pull request merged**